cdc/schema/storage.go

```diff
@@ -103,6 +103,41 @@ func (ti *TableInfo) GetIndexInfo(indexID int64) (info *model.IndexInfo, exist b
 	return ti.Indices[indexOffset], true
 }
 
+// WritableColumns returns all public and non-generated columns
+func (ti *TableInfo) WritableColumns() []*model.ColumnInfo {
+	cols := make([]*model.ColumnInfo, 0, len(ti.Columns))
+	for _, col := range ti.Columns {
+		if col.State == model.StatePublic && !col.IsGenerated() {
+			cols = append(cols, col)
+		}
+	}
+	return cols
+}
+
+// GetUniqueKeys returns all unique keys of the table as a slice of column names
+func (ti *TableInfo) GetUniqueKeys() [][]string {
+	var uniqueKeys [][]string
+	for _, idx := range ti.Indices {
+		if idx.Primary || idx.Unique {
+			colNames := make([]string, 0, len(idx.Columns))
+			for _, col := range idx.Columns {
+				colNames = append(colNames, col.Name.O)
+			}
+			uniqueKeys = append(uniqueKeys, colNames)
+		}
+	}
+	if ti.PKIsHandle {
+		for _, col := range ti.Columns {
+			if mysql.HasPriKeyFlag(col.Flag) {
+				// Prepend to make sure the primary key ends up at the front
+				uniqueKeys = append([][]string{{col.Name.O}}, uniqueKeys...)
+				break
+			}
+		}
+	}
```

Comment:
should we make sure that the `pk` is the first in unique keys?

Reply:
I'm not sure, just to make it compatible with the original `inspector` implementation.